test/test_ops.py

```diff
@@ -193,6 +226,48 @@ def wrapped_fn(*args, **kwargs):
 
             self.assertEqual(result, expected)
 
+    @ops(functorch_lagging_op_db + additional_op_db, allowed_dtypes=(torch.float,))
+    @skipOps('TestOperators', 'test_jvp', set({
+        # These are issues that should be fixed in core. See repro in core:
+        # https://github.com/pytorch/functorch/pull/232#discussion_r751405155
+        # RuntimeError: expected scalar type double but found float
+        xfail('minimum'),
+        xfail('min', 'binary'),
+        xfail('maximum'),
+        xfail('max', 'binary'),
+
+        # Composite ops that do bad things
+        # RuntimeError: Cannot access data pointer of Tensor that doesn't have storage
+        xfail('linalg.inv'),
+        xfail('linalg.matrix_power'),
+        xfail('linalg.cholesky'),
+        xfail('tensor_split'),
+
+        # Supposed to support forward AD, but actually doesn't
+        xfail('linalg.eigvalsh'),
+        xfail('var_mean'),
+        xfail('std_mean'),
+    }))
+    def test_jvp(self, device, dtype, op):
+        # TODO: when we change supports_autograd to supports_backward_ad, also change in this file
+        if not op.supports_forward_ad:
+            self.skipTest("Skipped! Forward AD not supported.")
+            return
+
+        samples = op.sample_inputs(device, dtype, requires_grad=True)
+        # TODO: test in-place
+        if is_inplace(op, op.get_op()):
+            self.skipTest("Skipped! NYI: inplace-testing not supported.")
+            return
+
+        for sample in samples:
+            fn, primals = normalize_op_input_output(op, sample, is_jvp=True)
+            tangents = tree_map(lambda x: torch.randn_like(x), primals)
+            primal_outs, tangent_outs = jvp(fn, primals, tangents)
+            expected_primal_outs, expected_tangent_outs = ref_jvp(fn, primals, tangents)
+            self.assertEqual(_as_tuple(primal_outs), expected_primal_outs)
+            self.assertEqual(_as_tuple(tangent_outs), expected_tangent_outs)
```

Comment:
This check is a little dangerous. If primals_out is a single Tensor, but functorch.jvp is wrong and always returns Tuples of Tensors instead of a single Tensor, we would not catch it!

Reply:
Good catch, I guess I just needed to update the reference implementation to not return a tuple when the size of the tuple is 1.